thread.c

```diff
@@ -545,9 +545,125 @@ static void thread_libevent_process(int fd, short which, void *arg) {
     }
 }
 
-/* Which thread we assigned a connection to most recently. */
+/* The thread we assigned a connection to most recently. */
 static int last_thread = -1;
 
+/* This function simply returns next_thread based on round-robin */
+static LIBEVENT_THREAD *select_next_thread(void)
+{
+    int tid = (last_thread + 1) % settings.num_threads;
+    last_thread = tid;
+    return threads + tid;
+}
+
+/* This function selects a thread to process requests from a given connection.
+ * Thread selection is either based on the (existing) default mechanism,
+ * round-robin based thread selection, or based on NAPI_ID. NAPI_ID based
+ * selection creates an association between a (hardware) queue on a NIC and an
+ * application thread. This mapping streamlines the flow of data from the NIC
+ * to an application thread by having an application thread source data from
+ * the same hardware queue which serves up traffic for many connections
+ */
+static void select_thread(int sfd, LIBEVENT_THREAD **thread_selected)
+{
+    LIBEVENT_THREAD *thread, *selected_thread = NULL;
+    socklen_t len;
+    int napi_id;
+
+    /* if napi-id based thread selection is not enabled, use
+     * default round-robin based thread selection
+     */
+	if (!settings.napi_id_based_thread_selection) {
+        selected_thread = select_next_thread();
+        goto out;
```

Comment:
I might be wrong but the only uses of goto in the code are for error handling; or should be almost all of it.

might be okay with this case though..

Reply:
I will rewrite this function and eliminate goto